internal/db/db.go

```diff
@@ -4,151 +4,15 @@ import (
 	"context"
 	"fmt"
 	"io"
-	"os"
-	"regexp"
-	"strconv"
-	"strings"
 
-	"github.com/jackc/pgx/v5/pgconn"
 	"github.com/jackc/pgx/v5/stdlib"
 	"github.com/jmoiron/sqlx"
 )
 
-// minimalConnConfig is the minimal settings we need for connection. More
-// unusual options are currently not supported.
-type minimalConnConfig struct {
-	host     string
-	user     string
-	password string
-	db       string
-	port     uint16
-	sslmode  string
-}
-
-// DSN returns the PostgreSQL compatible DSN string that corresponds to mcc.
-// This is expressed as a string of <key>=<value> separated by spaces.
-func (mcc *minimalConnConfig) DSN() string {
-	var s strings.Builder
-	writeNonempty := func(key, val string) {
-		if val != "" {
-			_, err := s.WriteString(key + "=" + val + " ")
-			if err != nil {
-				panic(err)
-			}
-		}
-	}
-	writeNonempty("host", mcc.host)
-	writeNonempty("user", mcc.user)
-	writeNonempty("password", mcc.password)
-	writeNonempty("dbname", mcc.db)
-	if mcc.port != 0 {
-		writeNonempty("port", strconv.FormatUint(uint64(mcc.port), 10))
-	}
-	writeNonempty("sslmode", mcc.sslmode)
-	writeNonempty("application_name", "timescaledb-parallel-copy")
-
-	return strings.TrimSpace(s.String())
-}
-
-// Overrideable is an interface for defining ways to override PG settings
-// outside of the usual manners (through the connection string/URL or env vars).
```

Comment:
@MetalBlueberry pgx supports both the key=value and URI right? I'm like 99% sure, just want to double check.

Reply:
https://github.com/jackc/pgx/blob/672c4a3a24849b1f34857817e6ed76f6581bbe90/pgconn/config.go#L161

here it is the documentation. It does